### PR TITLE
[6.1][BUGS#1191] fix: jackson custom serializer for ProjectFileStorage

### PR DESCRIPTION
**Pull request by miurahr**
ProjectFileStorage class is use to read and write `omegat.project` file.
This PR change Jackson object mapper configuration using custom serializer for `otherAttributes` attribute.
Current master branch produce `<otherAttributes/>` node in `omegat.project` when there is no attribute, and fix it with custom selializer.

I've added test cases that is as same as `releases/6.0`, so it guarantee a compaitibility between versions.

## Pull request type

- Bug fix -> [bug]

## Which ticket is resolved?
 
- ProjectFile has unwanted XML element otherAttributes
  * https://sourceforge.net/p/omegat/bugs/1191/

## What does this PR change?

- Add test as same as  #662
- Add custom serializer for `Map<QName, String> otherAttributes` object.

### CODE REVIEW

File: src/org/omegat/util/ProjectOtherAttributesSerializer.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.databind.type.MapType;
+
+public class ProjectOtherAttributesSerializer extends StdSerializer<Map<QName, String>> {
```
Comment (miurahr) [inline src/org/omegat/util/ProjectOtherAttributesSerializer.java:38]: Can we move the serializer as internal class?
It is only used in ProjectFileStorage class and not considered for extension and used in another class.

File: test/src/org/omegat/util/ProjectFileStorageTest.java [hunk 2]
```diff
@@ -58,11 +68,19 @@ public final void setUp() throws Exception {
         tempDir = Files.createTempDirectory("omegat").toFile().getAbsoluteFile();
         assertTrue(tempDir.isDirectory());
         TestPreferencesInitializer.init(tempDir.getPath());
+        XMLUnit.setIgnoreWhitespace(true);
+        XMLUnit.setIgnoreAttributeOrder(true);
+        XMLUnit.setIgnoreComments(true);
+        // XMLUnit.setControlEntityResolver(null);
+        // XMLUnit.setTestEntityResolver(null);
```
Comment (miurahr) [inline test/src/org/omegat/util/ProjectFileStorageTest.java:75]:
```suggestion
```

File: test/src/org/omegat/util/ProjectFileStorageTest.java [hunk 3]
```diff
@@ -58,11 +68,17 @@ public final void setUp() throws Exception {
...
+        XMLUnit.setIgnoreAttributeOrder(true);
+        XMLUnit.setIgnoreComments(true);
+
     }
 
     @After
     public final void tearDown() throws Exception {
         FileUtils.deleteDirectory(tempDir);
+        XMLUnit.setControlEntityResolver(null);
+        XMLUnit.setTestEntityResolver(null);
```
Comment (miurahr) [inline test/src/org/omegat/util/ProjectFileStorageTest.java:81]:
```suggestion
```
No need to reset.

**Comment by brandelune** [comment]
Please:
- add a line to changes.txt

**Comment by miurahr** [comment]
> Please:
> 
> * add a line to changes.txt

It has already been.
https://github.com/omegat-org/omegat/blob/master/release/changes.txt#L61-L62

**Pull request merged**